### PR TITLE
Fix SAML integration tests

### PR DESCRIPTION
**Pull request by arturo-seijas**
Integration tests improvements:
* Change waiting status conditions to improve reliability
* Deploy edge revisions of the redis charm that fix some issues when setting up the state
* Clean rockcraft after image builds to prevent the runner to run out of disk

### CODE REVIEW

**Review by nrobinaubertin — DISMISSED** [review]

**Comment by jdkandersson** [comment]
Could a description be provided please?

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def patched_getaddrinfo(*args):
         csrf_token_matches = re.findall(
             "<input type='hidden' name='csrfmiddlewaretoken' value='([^']+)' />", login_page.text
         )
-        assert len(csrf_token_matches) > 0
+        assert len(csrf_token_matches), login_page.text
```
Comment (jdkandersson) [inline tests/integration/test_charm.py:253]: Could this just be `assert csrf_token_matches`?

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -80,10 +80,10 @@ async def app(
     """
     assert ops_test.model
     # Deploy relations to speed up overall execution
-    dependencies = asyncio.gather(
-        ops_test.model.deploy("postgresql-k8s", channel="latest/stable", series="focal"),
-        ops_test.model.deploy("redis-k8s", "redis-broker"),
-        ops_test.model.deploy("redis-k8s", "redis-cache"),
+    asyncio.gather(
+        ops_test.model.deploy("postgresql-k8s", channel="latest/edge", series="focal"),
+        ops_test.model.deploy("redis-k8s", "redis-broker", channel="latest/edge"),
```
Comment (amandahla) [inline tests/integration/conftest.py:85]: Out of curiosity: Why was changed to edge channel? Does the stable one not being updated anymore?

Comment (arturo-seijas) [reply tests/integration/conftest.py:85]: The stable versions have some stablity issues, specially redis

File: tests/integration/test_charm.py (hunk 1 above)
Comment (arturo-seijas) [reply tests/integration/test_charm.py]: Yes, thx

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by merkata — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by merkata — DISMISSED** [review]

**Review by weiiwang01 — APPROVED** [review]

**Review by gtrkiller — APPROVED** [review]
LGTM

**Review by merkata — APPROVED** [review]
Great!

**Pull request merged**